From 69a3ba6190b586a25d1422ce856b5980273331ab Mon Sep 17 00:00:00 2001
From: Marcel Huber <marcelhuberfoo@gmail.com>
Date: Mon, 24 Nov 2014 22:28:42 +0100
Subject: spl mkconf patch


diff --git scripts/dkms.mkconf scripts/dkms.mkconf
index caba1be..2b60e2b 100755
--- scripts/dkms.mkconf
+++ scripts/dkms.mkconf
@@ -4,31 +4,33 @@ PROG=$0
 
 pkgcfg=/etc/sysconfig/zfs
 
-while getopts "n:v:c:f:" opt; do
+while getopts "n:v:c:f:s:" opt; do
 	case $opt in
 		n) pkgname=$OPTARG ;;
 		v) pkgver=$OPTARG  ;;
+		s) pkgverspl=$OPTARG  ;;
 		c) pkgcfg=$OPTARG ;;
 		f) filename=$OPTARG ;;
 	esac
 done
 
 if [ -z "${pkgname}" -o -z "${pkgver}" -o -z "${filename}" ]; then
-	echo "Usage: $PROG -n <pkgname> -v <pkgver> -c <pkgcfg> -f <filename>"
+	echo "Usage: $PROG -n <pkgname> -v <pkgver> [-s <pkgverspl>] -c <pkgcfg> -f <filename>"
 	exit 1
 fi
 
 cat >${filename} <<EOF
 PACKAGE_NAME="${pkgname}"
 PACKAGE_VERSION="${pkgver}"
+PACKAGE_VERSION_SPL="${pkgverspl:-$pkgver}"
 PACKAGE_CONFIG="${pkgcfg}"
 PRE_BUILD="configure
   --prefix=/usr
   --with-config=kernel
   --with-linux=\${kernel_source_dir}
   --with-linux-obj=\${kernel_source_dir}
-  --with-spl=\${source_tree}/spl-\${PACKAGE_VERSION}
-  --with-spl-obj=\${dkms_tree}/spl/\${PACKAGE_VERSION}/\${kernelver}/\${arch}
+  --with-spl=\${source_tree}/spl-\${PACKAGE_VERSION_SPL}
+  --with-spl-obj=\${dkms_tree}/spl/\${PACKAGE_VERSION_SPL}/\${kernelver}/\${arch}
   --with-spl-timeout=300
   \$(
     [[ -r \${PACKAGE_CONFIG} ]] \\
-- 
2.1.3

