From 2a628b21af3bb36e5aa1acdeb6ba37224de1e809 Mon Sep 17 00:00:00 2001
From: Marcel Huber <marcelhuberfoo@gmail.com>
Date: Wed, 5 Oct 2016 22:31:11 +0200
Subject: add new icp module to dkms config

---
 scripts/dkms.mkconf | 4 ++++
 1 file changed, 4 insertions(+)

diff --git scripts/dkms.mkconf scripts/dkms.mkconf
index caba1be..ad97f9b 100755
--- scripts/dkms.mkconf
+++ scripts/dkms.mkconf
@@ -70,6 +70,7 @@ STRIP[2]="\${STRIP[0]}"
 STRIP[3]="\${STRIP[0]}"
 STRIP[4]="\${STRIP[0]}"
 STRIP[5]="\${STRIP[0]}"
+STRIP[6]="\${STRIP[0]}"
 BUILT_MODULE_NAME[0]="zavl"
 BUILT_MODULE_LOCATION[0]="module/avl/"
 DEST_MODULE_LOCATION[0]="/extra/avl/avl"
@@ -88,4 +89,7 @@ DEST_MODULE_LOCATION[4]="/extra/zfs/zfs"
 BUILT_MODULE_NAME[5]="zpios"
 BUILT_MODULE_LOCATION[5]="module/zpios/"
 DEST_MODULE_LOCATION[5]="/extra/zpios/zpios"
+BUILT_MODULE_NAME[6]="icp"
+BUILT_MODULE_LOCATION[6]="module/icp/"
+DEST_MODULE_LOCATION[6]="/extra/icp/icp"
 EOF
-- 
2.10.0

